src/org/intellij/erlang/psi/impl/ErlangFileImpl.java

```diff
@@ -712,6 +719,27 @@ public boolean process(PsiElement psiElement) {
 
   @NotNull
   @Override
+  public Collection<ErlangCallbackFunction> getOptionalCallbacks() {
+    return myOptionalCallbacks.getValue();
+  }
+
+  private Collection<ErlangCallbackFunction> calcOptionalCallbacks() {
+    ErlangFileStub stub = getStub();
+    if (stub != null) {
+      return getChildrenByType(stub, ErlangTypes.ERL_CALLBACK_FUNCTION, new ArrayFactory<ErlangCallbackFunction>() {
+        @NotNull
+        @Override
+        public ErlangCallbackFunction[] create(int count) {
+          return new ErlangCallbackFunction[count];
+        }
+      });
```

Comment:
Please, preserve code style.  All other ```calc***``` methods here reference corresponding stub type's ARRAY_FACTORY field. Use the one in org.intellij.erlang.stubs.types.ErlangCallbackStubElementType. Not only you'll reuse existing object, but you'll also improve your code readability.